skopt/optimizer/base.py

```diff
@@ -220,33 +303,42 @@ def base_minimize(func, dimensions, base_estimator,
     # setting the scope for these variables
     result = None
 
-    # User suggested points at which to evaluate the objective first
-    if x0 and y0 is None:
-        y0 = list(map(func, x0))
-        n_calls -= len(y0)
+    # TODO: I tried to clean this up a little, but it is VERY confusing.
+    # Pass user suggested initialisation points to the optimizer.
+    if x0 is not None:
```

Comment:
`x0` should already be initialised as an empty list by the time we get here, so we should keep the old test (`if x0`).

I like the comment but splitting up the if clause makes it harder to understand

Reply:
As I recall there was an `if x0:` statement here. That is not Python, that is C-style programming! And it can actually fail in Python if the object has overrided some dunder-method (I can't remember which one but I think there's a PyCon talk about this ...) In Python you should make it explicit what you are testing against e.g. by writing `if x0 is not None:`

I think I'll re-submit the wrapper function as discussed below and I'll remove all these changes to the other files and just add the wrapper-function with as few changes as possible. It would be nice if you would go over some of these files and polish them.